core/Enclave/enclave_hsm.edl

```diff
@@ -68,48 +68,67 @@ enclave {
-                [out, size=cipherblob_len] uint8_t *cipherblob, size_t cipherblob_len);
+                                                [in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
+                                                [in, size=plaintext_len] const uint8_t *plaintext, size_t plaintext_len,
+                                                [out, size=cipherblob_len] uint8_t *cipherblob, size_t cipherblob_len,
+                                                ehsm_keyspec_t keyspec, uint32_t key_size);
 
         public sgx_status_t enclave_aes_decrypt([in, size=aad_len] const uint8_t *aad, size_t aad_len,
-                [in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                [in, size=cipherblob_len] const uint8_t *cipherblob, size_t cipherblob_len,
-                [out, size=plaintext_len] uint8_t *plaintext, size_t plaintext_len);
-
-        public sgx_status_t enclave_generate_datakey(uint32_t key_spec,
-                [in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                [in, size=context_len] const uint8_t *context, size_t context_len,
-                [out, size=plain_key_len] uint8_t *plain_key, size_t plain_key_len,
-                [out, size=encrypted_key_len] uint8_t *encrypted_key, size_t encrypted_key_len);
-
-        public sgx_status_t enclave_export_datakey(uint32_t cmk_spec,
-                [in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                [in, size=context_len] const uint8_t *context, size_t context_len,
-                [in, size=encrypted_key_len] uint8_t *encrypted_key, size_t encrypted_key_len,
-				uint32_t uk_spec,
-				[in, size=uk_blob_size] const uint8_t *uk_blob, size_t uk_blob_size,
-                [out, size=new_encrypted_key_len] uint8_t *new_encrypted_key, size_t new_encrypted_key_len);
-
-        public sgx_status_t enclave_create_rsa_key([out, size=cmk_blob_size] uint8_t *cmk_blob, uint32_t cmk_blob_size,
-                                               [out] uint32_t *req_blob_size);
+                                                [in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
+                                                [in, size=cipherblob_len] const uint8_t *cipherblob, size_t cipherblob_len,
+                                                [out, size=plaintext_len] uint8_t *plaintext, size_t plaintext_len,
+                                                ehsm_keyspec_t keyspec, uint32_t key_size);
+
         public sgx_status_t enclave_rsa_sign([in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                                         [in, size=data_len] const uint8_t *data, uint32_t data_len,
-                                         [out, size=signature_len] uint8_t *signature, uint32_t signature_len);
+                                         uint8_t padding_mode, uint32_t digest_mode, uint8_t keyspec,
```

Comment:
not sure, for example:
enclave_rsa_sign, enclave_ecc_sign, enclave_sm2_sign -> enclave_sign

enclave_aes_encrypt, enclave_sm4_encrypt -> enclave_encrypt
enclave_aes_decrypt, enclave_sm4_decrypt -> enclave_decrypt